### PR TITLE
misc urlutil updates

### PR DESCRIPTION
**Pull request by tarunKoyalwar**
### Proposed Changes

- Based on & to be merged in #286 
- remove globals instead create seperated methods
- more documentation on available parsing methods 
- minor refactor to remove duplicated code
Screenshot 2023-12-06 at 12 30 27 AM

### CODE REVIEW

**Comment by tarunKoyalwar** [comment]
## follow up

- merge this branch in https://github.com/projectdiscovery/utils/pull/286
- and merge that PR as logic / implementation was correct ( this PR is overall refactor + some more documentation on parsing methods)

File: url/parsers.go [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+// Parse and return URL (can be relative or absolute)
+func ParseURL(inputURL string, unsafe bool) (*URL, error) {
+	u := &URL{
+		URL:      &url.URL{},
+		Original: inputURL,
+		Unsafe:   unsafe,
+		Params:   NewOrderedParams(),
+	}
+	var err error
+	u, err = absoluteURLParser(u)
```
Comment (dogancanbakir) [inline url/parsers.go:35]: ParseAbsoluteURL*

Comment (dogancanbakir) [reply url/parsers.go:35]: Oh, the behavior is diff... In that case, please ignore

**Review by dogancanbakir — APPROVED** [review]
LGTM

**Pull request merged**